src/parser/shared/modules/HotTracker.ts

```diff
@@ -81,23 +89,24 @@ class HotTracker extends Analyzer {
     }
 
     let hotDuration = this.hotInfo[spellId].duration;
-    if (this.hotInfo[spellId].durationConditions) {
-      event.selectedCombatant = this.selectedCombatant;
-      hotDuration = this.hotInfo[spellId].durationConditions(event);
+    const durationConditionsFunc = this.hotInfo[spellId].durationConditions;
+    if (durationConditionsFunc !== undefined) {
+      const c: Combatant = this.selectedCombatant;
+      hotDuration = durationConditionsFunc(event, c);
     }
 
     let maxDuration = -1;
-    if (this.hotInfo[spellId].maxDuration) {
-      event.selectedCombatant = this.selectedCombatant;
-      maxDuration = this.hotInfo[spellId].maxDuration(event);
+    const maxDurationFunc = this.hotInfo[spellId].maxDuration;
+    if (maxDurationFunc !== undefined) {
+      const c: Combatant = this.selectedCombatant;
+      maxDuration = maxDurationFunc(event, c);
     }
 
     const newHot = {
       start: event.timestamp,
       end: event.timestamp + hotDuration,
       originalEnd: event.timestamp + hotDuration,
       spellId, // stored extra here so I don't have to convert string to number like I would if I used its key in the object.
-      guid: this.currentGuid,
```

Comment:
this was some sort of 'counting' functionality that was totally unused and served no clear purpose, hence why I removed it.

Reply:
This was more for debugging purposes than anything else as having to find a specific hot that is breaking this can be painful.

Reply:
ahh that makes sense - maybe I'll find myself adding it again some day